File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        result = nums
        
        for i in range(len(result)-1):
            if result[i] == result[i+1]:
                result[i] *= 2
                result[i+1] = 0
        
        result = list(filter(lambda x: x > 0, result))
        
        zeroes = [0] * (len(nums) - len(result))
        
        return result + zeroes

File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.swift
class Solution {
    func applyOperations(_ nums: [Int]) -> [Int] {
        var nums = nums
        
        for i in 0..<nums.count - 1 {
            if nums[i] == nums[i+1] {
                nums[i] *= 2
                nums[i+1] = 0
            }
        }
        
        var result = nums.filter { $0 > 0 }
        var zeroes = Array(repeating: 0, count: nums.count - result.count)
        result.append(contentsOf: zeroes)
        
        return result
    }
}